code/modules/games/cards.dm

```diff
@@ -0,0 +1,469 @@
+	if(user==target)
+		user.visible_message("<span class = 'notice'>[user] deals [dcard] card(s) to \himself.</span>")
+	else
+		user.visible_message("<span class = 'notice'>[user] deals [dcard] card(s) to [target].</span>")
+	H.throw_at(get_step(target,target.dir),10,1,H)
+
+
+/obj/item/weapon/hand/attackby(obj/O as obj, mob/user as mob)
+	if(cards.len == 1 && istype(O, /obj/item/weapon/pen))
+		var/datum/playingcard/P = cards[1]
+		if(P.name != "Blank Card")
+			to_chat(user,"<span class = 'notice'>You cannot write on that card.</span>")
+			return
+		var/cardtext = sanitize(input(user, "What do you wish to write on the card?", "Card Editing") as text|null, MAX_PAPER_MESSAGE_LEN)
+		if(!cardtext)
+			return
+		P.name = cardtext
+		// SNOWFLAKE FOR CAG, REMOVE IF OTHER CARDS ARE ADDED THAT USE THIS.
+		P.card_icon = "cag_white_card"
+		update_icon()
+	else if(istype(O,/obj/item/weapon/hand))
+		var/obj/item/weapon/hand/H = O
+		if(H.parentdeck == src.parentdeck)
+			for(var/datum/playingcard/P in cards)
+				H.cards += P
+			H.concealed = src.concealed
+			qdel(src)
+			H.update_icon()
+			return
+		else
+			to_chat(user,"<span class = 'notice'>You cannot mix cards from other deck!</span>")
+			return
+	..()
+
+/obj/item/weapon/deck/attack_self()
+	deckshuffle()
+
+/obj/item/weapon/deck/verb/verb_shuffle()
+	set category = "Object"
+	set name = "Shuffle"
+	set desc = "Shuffle the cards in the deck."
+	set src in view(1)
+	deckshuffle()
+
+/obj/item/weapon/deck/proc/deckshuffle()
+	var/mob/living/user = usr
+	if (cooldown < world.time - 50)
+		cards = shuffle(cards)
+		user.visible_message("<span class = 'notice'>[user] shuffles [src].</span>")
+		playsound(user, 'sound/items/cardshuffle.ogg', 50, 1)
+		cooldown = world.time
+
+
+/obj/item/weapon/deck/MouseDrop(atom/over_object) // Code from Paper bin, so you can still pick up the deck
+	var/mob/M = usr
+	if(M.incapacitated() || !Adjacent(M))
+		return
+
+	if(over_object == M)
+		M.put_in_hands(src)
+
+	else if(istype(over_object, /obj/screen))
+		switch(over_object.name)
+			if("r_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(src)
+				M.put_in_r_hand(src)
+			if("l_hand")
+				if(!remove_item_from_storage(M))
+					M.unEquip(src)
+				M.put_in_l_hand(src)
+
+	add_fingerprint(M)
+	usr.visible_message("<span class = 'notice'>[usr] pick up the deck.</span>")
+
+/obj/item/weapon/pack
+	name = "Card Pack"
+	desc = "For those with disposable income."
+
+	icon_state = "card_pack"
+	icon = 'icons/obj/playing_cards.dmi'
+	w_class = WEIGHT_CLASS_TINY
+	var/list/cards = list()
+	var/parentdeck = null // For future card pack that need to be compatible with eachother i.e. cardemon
+
+
+/obj/item/weapon/pack/attack_self(mob/user as mob)
+	user.visible_message("<span class = 'notice'>[name] rips open the [src]!</span>")
+	var/obj/item/weapon/hand/H = new()
+
+	H.cards += cards
+	cards.Cut();
+	user.drop_item()
+	qdel(src)
+
+	H.update_icon()
+	user.put_in_active_hand(H)
+
+/obj/item/weapon/hand
```

Comment:
This seems like a typepath with a really big chance of a namespace conflict. Would it be better to put all the types you're declaring here under the `/obj/item/weapon/cards` type?